webui/react/src/components/kit/LogViewer/LogViewerSelect.tsx

```diff
@@ -86,31 +89,61 @@ const LogViewerSelect: React.FC<Props> = ({
 
   const handleChange = useCallback(
     (key: keyof Filters, caster: NumberConstructor | StringConstructor) => (value: SelectValue) => {
-      onChange?.({ ...values, [key]: (value as Array<string>).map((item) => caster(item)) });
+      setFilters({ ...values, [key]: (value as Array<string>).map((item) => caster(item)) });
     },
-    [onChange, values],
+    [values],
   );
 
   const handleSearch = useCallback(
     (e: React.ChangeEvent<HTMLInputElement>) =>
-      onChange?.({ ...values, searchText: e.target.value }),
-    [onChange, values],
+      setFilters({ ...values, searchText: e.target.value }),
+    [values],
+  );
+
+  const handleReset = useCallback(() => {
+    setFilters({});
+    onReset?.();
+  }, [onReset]);
+
+  const throttledChangeFilter = useMemo(
+    () =>
+      throttle(
+        500,
+        (f: Filters) => {
+          onChange?.(f);
+        },
+        { noLeading: true },
+      ),
+    [onChange],
   );
 
-  const handleReset = useCallback(() => onReset?.(), [onReset]);
+  useEffect(() => {
+    if (!filters) return;
+    throttledChangeFilter(filters);
+  }, [filters, throttledChangeFilter]);
+
+  useEffect(() => {
+    return () => {
+      throttledChangeFilter.cancel();
+    };
+  }, [throttledChangeFilter]);
 
   return (
     <>
       <Space>
         {showSearch && (
-          <Input placeholder="Search Logs..." value={values.searchText} onChange={handleSearch} />
+          <Input
+            placeholder="Search Logs..."
+            value={filters.searchText || values.searchText}
+            onChange={handleSearch}
+          />
         )}
         {moreThanOne.allocationIds && (
           <Select
             disableTags
             mode="multiple"
             placeholder={`All ${LABELS.allocationIds}`}
-            value={values.allocationIds}
+            value={filters.allocationIds || values.allocationIds}
```

Comment:
question: do we need `|| values.someVal`?

Reply:
This was to show previously stored value in setting when page first mounts, but initialize `filters` with `values` has the same effect